src/main/scala/li/cil/oc/server/fs/FileInputStreamFileSystem.scala

```diff
@@ -38,7 +38,7 @@ trait FileInputStreamFileSystem extends InputStreamFileSystem {
     case file if file.exists() && file.isFile => Array(file.getName)
     case directory if directory.exists() && directory.isDirectory && directory.list() != null =>
       directory.listFiles().map(file => if (file.isDirectory) file.getName + "/" else file.getName)
-    case _ => throw new io.FileNotFoundException("no such file or directory")
+    case _ => throw new io.FileNotFoundException(path + " (no such file or directory)")
```

Comment:
How about changing the format to `"no such file or directory: " + path`? That way you can see the failure reason right away even if the path is very long, which I find more useful than putting the error message after the path (where it's easy to miss too). That's how Python does it for example:

```python
>>> open("/usr/bin/nope")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: '/usr/bin/nope'
```

Same goes for `"read-only filesystem"` down below.